### PR TITLE
Lift transitive dependency via patchconfig

### PR DESCRIPTION
**Pull request by Tratcher**
https://github.com/dotnet/aspnetcore/issues/18265 @JunTaoLuo is this enough to lift the transitive dependencies?

The 2.1.14 patch shipped the correct packages containing the necessary code changes, but it did not update the transitive dependency graph so that the updated dependencies would be pulled in. E.g. most projects only reference Microsoft.AspNetCore.CookiePolicy, but updating that package did not automatically update Microsoft.Net.Http.Headers because the intermediate package Microsoft.AspNetCore.Http was not modified.

Impact: This primarily affects .NET Framework apps where ASP.NET Core is consumed as packages rather than a shared runtime. Customers only get one of the needed package updates and do not get the expected behavior.

Workaround: Add a direct dependency for the latest version of Microsoft.Net.Http.Headers.

Risk: low

### CODE REVIEW

**Comment by anurse** [comment]
Approved for Feb. Pending review sign-off and undrafting. This needs to be merged 1/15 to make the build.

File: eng/PatchConfig.props [hunk 1]
```diff
@@ -43,11 +43,18 @@ Later on, this will be checked using this condition:
       Microsoft.Net.Http.Headers;
       Microsoft.AspNetCore.CookiePolicy;
     </PackagesInPatch>
-  </PropertyGroup>
-  <PropertyGroup Condition=" '$(VersionPrefix)' == '2.1.16' ">
+  </PropertyGroup>
```
Comment (BrennanConroy) [inline eng/PatchConfig.props]: nit:
```suggestion
  </PropertyGroup>
```

**Comment by anurse** [comment]
Undraft?

**Comment by anurse** [comment]
Oops, missed it. Icon still showed as draft in my notifications :).

**Review by JunTaoLuo — APPROVED** [review]

**Comment by Tratcher** [comment]
Looking at the build artifacts, this change didn't have any effect, the produced packages are all still referencing 2.1.1 dependencies. Looking at it with @JunTaoLuo

**Comment by JunTaoLuo** [comment]
After looking over the logic in this area, I'm unsure if this is supposed to work. Specifically: https://github.com/dotnet/aspnetcore/blob/release/2.1/eng/targets/ResolveReferences.targets#L44 says
> We don't use project references between components in servicing builds between compontents to preserve the baseline as much as possible.

It may be the case that we simply expect all users to reference all required packages individually. 

This also aligns with what we expect in terms of keeping the LZMA as small as possible back when it was a concern. Otherwise, we would be including many more assemblies in the LZMA of each patch. I'm not sure if that's something we're still concerned about but it at least makes sense why we build against the baseline. I'm more inclined to say that it's by design and we shouldn't change the way we patch now in 2.1.

cc @dougbu to double check my understanding here.

**Comment by Tratcher** [comment]
Hmm, that makes for a poor experience with these specific patches. @anurse ?

**Comment by dougbu** [comment]
If I'm reading this correctly, PatchConfig.props controls only what ships and not dependency versions i.e. projects build against baseline projects unconditionally, If that's correct, @JunTaoLuo's idea of the intent is likely correct: We were aiming to minimize what gets shipped in a patch -- especially w.r.t. the LZMA size.

Why do you think LZMA is no longer an issue @JunTaoLuo? Let's confirm that prior to our meeting w/ the leads this afternoon.

/fyi @Pilchie @anurse @mkArtakMSFT @ajcvickers

**Comment by Tratcher** [comment]
Closing this out as it's not addressing the issue. We'll address this via announcements and docs instead.

**Comment by JunTaoLuo** [comment]
No particular reason. I'm just not sure what is still being deployed for 2.1 in addition to packages and whether LZMA is still a relevant scenario.

**Pull request closed without merging**